WhaleSpotting/Controllers/ApiDataController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Text;
+using System.Net;
+using RestSharp;
+using WhaleSpotting.Models.ApiModels;
+using WhaleSpotting.Models.DbModels;
+using WhaleSpotting.Services;
+using Microsoft.AspNetCore.Authorization;
+
+namespace WhaleSpotting.Controllers
+{
+    [ApiController]
+    [Route("/getapidata")]
+    public class ApiDataController : ControllerBase
+    {
+        private readonly ISightingsService _sightings;
+
+        public ApiDataController(ISightingsService sightings)
+        {
+            _sightings = sightings;
+        }
+
+        [HttpPost("")]
+        public async Task<OkResult> GetApiData()
```

Comment:
I'd prefer this be with the IActionResult
```suggestion
        public async Task<IActionResult> GetApiData()
```

Reply:
:)